### PR TITLE
#238 When calculating payouts, take all previous months into account.

### LINKED ISSUE #238: No payouts for previous month's contributions
When a new month is created, and there have been contributions in previous months, we should be able to see the payouts for the previous months.

They aren't showing up which means that the payouts calculations probably aren't happening correctly.

### PR DESCRIPTION
**Pull request by Alfredo-Delgado**
Resolves #238

### CODE REVIEW

**Comment by justinbmeyer** [comment]
Thanks for working on this, but this isn't what I had in mind. The previous contribution months should be loaded to calculate the pot.

**Comment by Alfredo-Delgado** [comment]
Ok, I believe this is work that has not been addressed, yet. It'd be related to #161

**Comment by justinbmeyer** [comment]
I think the best short-term fix is to simply load every contribution month.  We probably need to also load the contributors to check ones that are active.

**Comment by Alfredo-Delgado** [comment]
The calculations already took previous months into account, although they don't exclude contributors whose `isActive` bit is false. This PR gathers all unique contributors and projects up to the current month before calculating the payouts.

File: public/components/payouts/payouts.js [hunk 1]
```diff
@@ -6,88 +6,92 @@ import moment from "moment";
 
 export const ViewModel = DefineMap.extend({
   // Passed properties
-  contributionMonth: ContributionMonth,
-  contributionMonthsPromise: {
-    value() {
-      return ContributionMonth.getList({});
-    }
+  contributionMonth: {
```
Comment (justinbmeyer) [inline public/components/payouts/payouts.js]: `contributionMonth: ContributionMonth`

File: public/models/contribution-month/contribution-month.js [hunk 2]
```diff
@@ -266,6 +264,109 @@ ContributionMonth.List = DefineList.extend({
...
+   */
+  getMonthlyPayouts(currentMonth) {
+		let currentMoment = moment(currentMonth.date);
+    let contributorProjectPayouts = {};
+    let uniqueContributors = {};
+    let uniqueProjects = {};
+
+    this.filter(contributionMonth => {
+      return currentMoment.isSameOrAfter(contributionMonth.date);
+    }).map(contributionMonth => {
```
Comment (justinbmeyer) [inline public/models/contribution-month/contribution-month.js]: `forEach`

File: public/models/contribution-month/contribution-month.js [hunk 3]
```diff
@@ -266,6 +264,109 @@ ContributionMonth.List = DefineList.extend({
...
+   *     ]
+   *   }
+   * }
+   * ```
+	 *
+	 * @param {ContributionMonth} currentMonth
+	 * @return {Map}
+   */
+  getMonthlyPayouts(currentMonth) {
+		let currentMoment = moment(currentMonth.date);
```
Comment (justinbmeyer) [inline public/models/contribution-month/contribution-month.js]: space issue?

**Comment by Alfredo-Delgado** [comment]
- replaced `map` with `forEach`
- replaced all spaces with tabs in `contribution-month.js`
- renamed cloned contributor to `contributorPayout` for better readability

**Pull request merged**